### ISSUE TITLE
travis36minver.txt needs an update

### ISSUE THREAD

**Issue by Kojoley** [matplotlib/matplotlib#16116]
Travis Python 3.6 job log:
```
ERROR: packaging 19.2 has requirement pyparsing>=2.0.2, but you'll have pyparsing 2.0.1 which is incompatible.
ERROR: pandas 0.25.3 has requirement python-dateutil>=2.6.1, but you'll have python-dateutil 2.1 which is incompatible.
```
Those errors do not fail the installation (why?) and seems to had been unnoticed.

**Comment by timhoffm** [comment]
Currently:
~~~
ERROR: packaging 20.1 has requirement pyparsing>=2.0.2, but you'll have pyparsing 2.0.1 which is incompatible.
ERROR: tox 3.14.4 has requirement six<2,>=1.14.0, but you'll have six 1.11.0 which is incompatible.
ERROR: pandas 1.0.1 has requirement python-dateutil>=2.6.1, but you'll have python-dateutil 2.1 which is incompatible.
~~~

This is not failing due to a backward compatibility policy in pip (https://github.com/pypa/pip/issues/6969#issuecomment-531538703). Nevertheless we should create a consistent build environment.

**Comment by Kojoley** [comment]
It seems like we need to replace `pip` with either [`poetry`](https://github.com/python-poetry/poetry) or [`dephell`](https://github.com/dephell/dephell) for build dependencies installation.

**Comment by timhoffm** [comment]
As long as the tests still pass, I'm fine with waiting for the pip improvements. Probably we're not using the parts of the libraries that have increased dependency requirements.

**Comment by tacaswell** [comment]
I am very :-1: on bringing one of the environment managers into Matplotlib.  There is still too much thrashing in that space.

I am :+1: on bumping the min pyparsing to 2.0.3 (will open a PR, from reading the change log, 2.0.3 fixes a bunch of regressions from 2.0.2).  We last touched this dependency in https://github.com/matplotlib/matplotlib/pull/8754 .  I suspect that there is some critical bug fix in 2.0.2 (which is what made `packaging` pin there).  Oddly, the 2.0.1 tag is gone from github, but the files are still up on pypi.

We don't use tox, so :woman_shrugging: that it is un-happy.

We probably should pin the 3.6 pandas version back to a version that supports dateutil. We bumped the python-dateutil dep last in https://github.com/matplotlib/matplotlib/pull/9317 .

Our guidelines (https://matplotlib.org/devdocs/devel/min_dep_policy.html#python-dependencies) say to support at least 12 months, but only bump if we have to and I don't think we have any compelling reason to drag these versions forward.
